compiler/executable/insert/executable.rs

```diff
@@ -69,15 +76,27 @@ pub fn compile(
         source_span,
     )?;
 
+    #[cfg(debug_assertions)]
+    let mut variable_positions_tmp = variable_positions
+        .iter()
+        .filter(|(var, _)| {
+            !(variable_registry.get_variable_name(**var).is_none()
+                && variable_registry.get_variable_category(**var).unwrap() == VariableCategory::RoleType)
+        })
+        .map(|(k, v)| (k.clone(), v.clone()))
+        .collect::<HashMap<_, _>>();
+    #[cfg(not(debug_assertions))]
+    compile_error!("Remove the above");
+
```

Comment:
TODO: Rebase on Dmitrii's changes to see if anonymous role-types are excluded from the row returned by a match.